samples-final/AspNetCore-Luis-Dispatch-Bot/LuisDispatchBot.cs

```diff
@@ -0,0 +1,186 @@
+                            {
+                                intentsResult2.Add($"'{intent.Key}', score {intent.Value}");
+                            }
+                            await context.SendActivity($"Intents detected by the weather app: \n\n{string.Join("\n\n", intentsResult2)}");
+
+                            // list the entities
+                            entitiesList = new List<string>();
+                            foreach (var entity in recognizerResult.Entities)
+                            {
+                                if (!entity.Key.ToString().Equals("$instance"))
+                                {
+                                    entitiesList.Add($"{entity.Key}: {entity.Value.First}");
+                                }
+                            }
+
+                            if (entitiesList.Count > 0)
+                            {
+                                await context.SendActivity($"The following entities were found in the message:\n\n{string.Join("\n\n", entitiesList)}");
+                            }
+
+                            // Here, you can add code for calling the hypothetical weather service, passing in any entity information that you need
+
+                            break;
+                        case "none":
+                        // You can provide logic here to handle the known None intent (none of the above).
+                        // In this example we fall through to the QnA intent.
+                        case "q_faq":
+                            QnAMaker qnaMaker = new QnAMaker(this.qnaOptions);
+                            var messageActivity = context.Activity.AsMessageActivity();
+                            if (!string.IsNullOrEmpty(messageActivity.Text))
+                            {
+                                var results = await qnaMaker.GetAnswers(messageActivity.Text.Trim()).ConfigureAwait(false);
+                                if (results.Any())
+                                {
+                                    await context.SendActivity(results.First().Answer);
+                                }
+                                else
+                                {
+                                    await context.SendActivity("Couldn't find an answer in the FAQ.");
+                                }
+                            }
+                            break;
+                        default:
+                            // The intent didn't match any case, so just display the recognition results.
+                            await context.SendActivity($"Dispatch intent: {topIntent.Value.intent} ({topIntent.Value.score}).");
+
+                            break;
+                    }
+                }                
+
+            }
+            else if (context.Activity.Type is ActivityTypes.ConversationUpdate)
+            {
+                foreach (var newMember in context.Activity.MembersAdded)
```

Comment:
We need to factor this to a method in the framework somewhere and call it from places.